File: 백준/Bronze/29196. 소수가 아닌 수 2/소수가 아닌 수 2.cc
#include <cmath>
#include <iostream>
using namespace std;
int main(void) {
    int p = 1, q = 1;
    double k;
    cin >> k;
    while (abs((double)p / q - k) > pow(10, -6)) {
        if ((double)p / q > k)
            q += 1;
        else
            p += 1;
    }
    cout << "YES" << "\n";
    cout << p << ' ' << q;
    return 0;
}

File: 백준/Bronze/29196. 소수가 아닌 수 2/소수가 아닌 수 2.py
import sys
input=lambda:sys.stdin.readline().rstrip()
k=float(input())
p,q=1,1
while abs(p/q-k)>10**-6:
    if p/q>k:
        q+=1
    else:
        p+=1
print("YES")
print(p,q)